src/backends/kafka/config.rs

```diff
@@ -0,0 +1,116 @@
+use rdkafka::config::ClientConfig as RdKafkaConfig;
+use std::collections::HashMap;
+
+const DEFAULT_QUEUED_MAX_MESSAGE_KBYTES: u32 = 50_000;
+const DEFAULT_QUEUED_MIN_MESSAGES: u32 = 10_000;
+
+#[derive(Debug, Clone)]
+pub struct KafkaConfig {
+    config_map: HashMap<String, String>,
+}
+
+impl KafkaConfig {
+    pub fn new_config(
+        bootstrap_servers: Vec<String>,
+        override_params: Option<HashMap<String, String>>,
+    ) -> Self {
+        let mut config_map = HashMap::new();
+        config_map.insert("bootstrap.servers".to_string(), bootstrap_servers.join(","));
+
+        let config = Self { config_map };
+
+        apply_override_params(config, override_params)
+    }
+
+    pub fn new_consumer_config(
+        bootstrap_servers: Vec<String>,
+        group_id: String,
+        auto_offset_reset: String,
+        _strict_offset_reset: bool, // TODO: Implement this
+        override_params: Option<HashMap<String, String>>,
+    ) -> Self {
+        let mut config = KafkaConfig::new_config(bootstrap_servers, None);
+        config.config_map.insert("group.id".to_string(), group_id);
+        config
+            .config_map
+            .insert("enable.auto.commit".to_string(), "false".to_string());
+        config
+            .config_map
+            .insert("auto.offset.reset".to_string(), auto_offset_reset);
+        config.config_map.insert(
+            "queued.max.messages.kbytes".to_string(),
+            DEFAULT_QUEUED_MAX_MESSAGE_KBYTES.to_string(),
+        );
+        config.config_map.insert(
+            "queued.min.messages".to_string(),
+            DEFAULT_QUEUED_MIN_MESSAGES.to_string(),
+        );
```

Comment:
```
Hashmap::from([
     (key, value),
     (key, value),
     (key, value),
])

Reply:
The hashmap already exists before these lines though, we are not creating it here.